src/streaming/controllers/ScheduleController.js

```diff
@@ -177,11 +177,12 @@ function ScheduleController(config) {
         validateExecutedFragmentRequest();
 
         const isReplacement = replaceRequestArray.length > 0;
-        if (replacingBuffer || isNaN(lastInitQuality) || switchTrack || isReplacement ||
+        const lastInitializedQuality = lastInitializedRepresentationInfo ? lastInitializedRepresentationInfo.quality : NaN;
+        if (replacingBuffer || !lastInitializedRepresentationInfo || switchTrack || isReplacement ||
```

Comment:
Might be more intuitive to check isNaN(lastInitializedQuality)

Reply:
Fixed!